### ISSUE TITLE
Context deadline exceeded pulling a large/complex bucket

### ISSUE THREAD

**Issue by andrewxhill** [textileio/textile#448]
Pulling the docs bucket using the CLI just now I hit `Error! Context deadline exceeded`.

**Comment by andrewxhill** [comment]
more on this. the bucket init happened, so `.textile` exists in the folder. so rerunning will error. but none of the files were pulled. 

interestingly, if i also run `hub buck destroy` on the folder (without the files pulled). i get,

```
> Warning! This action cannot be undone. The bucket and all associated data will be permanently deleted.
Are you absolutely sure: y█
> Error! Not pinned or pinned indirectly
```

**Comment by andrewxhill** [comment]
if i rerun `destroy` i get, 

```
> Warning! This action cannot be undone. The bucket and all associated data will be permanently deleted.
Are you absolutely sure: y█
> Error! Instance not found
```

and i can confirm that it no longer is in the list of existing.

**Comment by sanderpick** [comment]
If you have all the files locally, can you zip and upload? We can use that to hopefully repro locally.

**Comment by andrewxhill** [comment]
yep, it's site.zip in here https://hub.textile.io/thread/bafk3aqwvwzqm4vbbiihc2iqndvvhuewths4jncbf5vzdv6lgopgsyfa/buckets/bafzbeidcuqlispjdtu42aesomtfjfhm2g7jxejfnszu5bo5nrp7wl666lm

**Comment by sanderpick** [comment]
I spent some time poking at this. I can't reproduce locally, but I can on staging. Likely just because the whole operation takes longer than 10 min (see below). 

The context is reused for all the list path and get file requests. Despite a context timeout of 1 day when pulling, the re-used context times out after exactly 10 minutes, always in a list path request (after many successful list path requests). Nowhere can I see a time out of 10 minutes being used... so, more investigation needed.
